tests/ui/iter_with_drain.rs

```diff
@@ -0,0 +1,51 @@
+#![warn(clippy::iter_with_drain)]
```

Comment:
Since we're using `span_lint_and_sugg` for generating the suggestion, this should also be tested with 

```suggestion
// run-rustfix
#![warn(clippy::iter_with_drain)]
```